src/registry/artifactory.rs

```diff
@@ -76,12 +90,13 @@ impl Registry for Artifactory {
             }
         );
 
-        let artifact_uri: Url = {
-            let mut url = self.0.url.clone();
+        let artifact_uri = {
+            let path = dependency.manifest.registry.path().to_owned();
```

Comment:
Ah i see why you give it the whole credentials struct. Could we still wrap it in an arc?